packages/sdk/src/container/types.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Logger, LogLevel, LogFunction } from "../util/logger";
+
+export interface Mapping {
+  [key: string]: any;
+}
+
+export type ComponentOption = Record<string, unknown>;
+
+export interface ComponentContainer {
+  resolve(componentName: string, version?: string): unknown;
+}
+
+export interface Component extends Mapping {
+  name: string;
+  version: string;
+  initialize(container: ComponentContainer, logger: Logger): void;
```

Comment:
what is our solution if initialize requires additional parameters (inputs)?

Reply:
I think we can add new function to ComponentContainer interface if that's a common scenario. The `logger` parameter is to ensure that the log message has the component name.